mantle/kola/harness.go

```diff
@@ -532,6 +532,26 @@ func runProvidedTests(tests map[string]*register.Test, patterns []string, multip
 		plog.Fatal(err)
 	}
 
+	flight, err := NewFlight(pltfrm)
+	if err != nil {
+		plog.Fatalf("Flight failed: %v", err)
+	}
+	defer flight.Destroy()
+	// Generate non-exclusive test wrapper (run multiple tests in one VM)
+	var nonExclusiveTests []*register.Test
+	for _, test := range tests {
+		if test.NonExclusive {
+			nonExclusiveTests = append(nonExclusiveTests, test)
+			delete(tests, test.Name)
+		}
+	}
+
+	if len(nonExclusiveTests) > 0 {
```

Comment:
Unrelated but: I wonder here if we should special-case the `len(nonExclusiveTests) == 1` case so that we don't do any wrapping at all to keep the output simpler when doing a focused run (i.e. `kola run my.single.test`). Anyway, can discuss that in a follow-up!

Reply:
The special casing makes sense and it would definitely make output easier to read. I guess the cost to consider here is that introducing more special casing into kola would make it more difficult to follow the logic (from a development perspective). If this isn't a concern here then adding the special casing should be straight forward.

Reply:
Agreed there's a slight complexity trade-off. Instead of this, let's do the `--disable-non-exclusive` suggested in https://github.com/coreos/coreos-assembler/pull/2506 instead.